File: composeApp/src/androidMain/kotlin/com/app/starleet/roomdb/LactateRepository.kt
package com.app.starleet.roomdb

import kotlinx.coroutines.flow.Flow

class LactateRepository(
    private val dao: LactateDao
) {

    val allScans: Flow<List<LactateScanEntity>> =
        dao.getAllScans()

    val last7Scans: Flow<List<LactateScanEntity>> =
        dao.getLast7Scans()

    suspend fun insertScan(
        value: Double,
        sweat: String,
        change: Double
    ) {
        dao.insertScan(
            LactateScanEntity(
                lactateValue = value,
                sweatStatus = sweat,
                changeFromLast = change,
                timestamp = System.currentTimeMillis()
            )
        )
    }

    suspend fun getLatest() = dao.getLatestScan()
}

File: iosApp/iosApp/ViewModel/LactateRepository.swift
//
//  LactateRepository.swift
//  iosApp
//
//  Created by KBS on 3/3/26.
//

import SwiftData
import AVFoundation

class LactateRepository {

    private let context: ModelContext

    init(context: ModelContext) {
        self.context = context
    }

    // Insert
    func insertScan(value: Double, sweat: String, change: Double) {
        let scan = LactateScan(
            lactateValue: value,
            sweatStatus: sweat,
            changeFromLast: change
        )

        context.insert(scan)
        try? context.save()
    }

    // Get All
    func getAllScans() -> [LactateScan] {
        let descriptor = FetchDescriptor<LactateScan>(
            sortBy: [SortDescriptor(\.timestamp, order: .reverse)]
        )
        return (try? context.fetch(descriptor)) ?? []
    }

    // Last 7
    func getLast7Scans() -> [LactateScan] {
        var descriptor = FetchDescriptor<LactateScan>(
            sortBy: [SortDescriptor(\.timestamp, order: .reverse)]
        )
        descriptor.fetchLimit = 7
        return (try? context.fetch(descriptor)) ?? []
    }

    // Latest
    func getLatestScan() -> LactateScan? {
        var descriptor = FetchDescriptor<LactateScan>(
            sortBy: [SortDescriptor(\.timestamp, order: .reverse)]
        )
        descriptor.fetchLimit = 1
        return try? context.fetch(descriptor).first
    }
}
